presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java

```diff
@@ -473,9 +473,14 @@ private boolean canProcessDictionary(Page page)
             Block inputHashBlock = page.getBlock(inputHashChannel.get());
             DictionaryBlock inputDataBlock = (DictionaryBlock) page.getBlock(channels[0]);
 
-            verify(inputHashBlock instanceof DictionaryBlock, "data channel is dictionary encoded but hash channel is not");
-            verify(((DictionaryBlock) inputHashBlock).getDictionarySourceId().equals(inputDataBlock.getDictionarySourceId()),
-                    "dictionarySourceIds of data block and hash block do not match");
+            if (!(inputHashBlock instanceof DictionaryBlock)) {
+                // data channel is dictionary encoded but hash channel is not
+                processDictionary = false;
```

Comment:
Why not just return?

Reply:
I didn't want to make diff bigger... if i return here, i should eliminate `processDictionary` local variable, making return in other places as well.

Reply:
I don't think you need to go that far, but I'm fine either way